code/datums/config/_config.dm

```diff
@@ -0,0 +1,126 @@
+#define CONFIG_FLAG_BOOL               BITFLAG(0)
+#define CONFIG_FLAG_NUM                BITFLAG(1)
+#define CONFIG_FLAG_TEXT               BITFLAG(2)
+#define CONFIG_FLAG_LIST               BITFLAG(3)
+#define CONFIG_FLAG_HAS_VALUE          BITFLAG(4)
+
+/proc/get_config_value(var/config_decl)
+	var/decl/config/config_option = GET_DECL(config_decl)
+	return config_option.value
+
+/proc/set_config_value(var/config_decl, var/new_value, var/defer_config_refresh = FALSE)
+
+	// Get our actual value (loading from text etc. may change data type)
+	var/decl/config/config_option = GET_DECL(config_decl)
+	if((config_option.config_flags & CONFIG_FLAG_NUM) && istext(new_value))
+		new_value = text2num(new_value)
+	else if((config_option.config_flags & CONFIG_FLAG_TEXT) && isnum(new_value))
+		new_value = num2text(new_value)
+
+	// Identical values, no point updating.
+	if(config_option.value == new_value)
+		return
+
+	// Actually set the value.
+	var/old_value = config_option.value
+	config_option.set_value(new_value)
+	if(config_option.value != old_value && !defer_config_refresh)
+		config_option.update_post_value_set()
+
+/proc/toggle_config_value(var/config_decl)
+	var/decl/config/config_option = GET_DECL(config_decl)
+	if(!(config_option.config_flags & CONFIG_FLAG_BOOL))
+		CRASH("Attempted to toggle non-boolean config entry [config_decl].")
+	set_config_value(config_decl, !config_option.value)
+	return config_option.value
+
+/decl/config
+	decl_flags = DECL_FLAG_MANDATORY_UID
+	var/config_description
+	var/value
+	var/default_value
+	var/config_flags = CONFIG_FLAG_HAS_VALUE
+	var/dump_to_file = "config/configuration.txt"
```

Comment:
Here's a suggestion:
What about maybe dividing configuration options by categories instead of putting a filename directly into every single one of them? And then the config handler would sort out what categories goes into what config file?

Reply:
Sounds good!